### PR TITLE
DBZ-6873 Support MySQL 8.2

### PR DESCRIPTION
**Pull request by ani-sha**
https://issues.redhat.com/browse/DBZ-6873

### CODE REVIEW

**Review by jpechane — CHANGES_REQUESTED** [review]
@ani-sha PLease take a look at test results in closer detail. There are almost 60 failures.

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java [hunk 1]
```diff
@@ -611,6 +611,15 @@ public static SecureConnectionMode parse(String value, String defaultValue) {
             .withDescription("The password for the trust store file. "
                     + "Used to check the integrity of the truststore, and unlock the truststore.");
 
+    public static final Field ALLOW_PUBLIC_KEY_RETRIEVAL = Field.create("database.allowPublicKeyRetrieval")
```
Comment (jpechane) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java:614]: I don't think wee need the config property for it. It is just `driver.` pass-thorugh property.

**Review by jpechane — COMMENTED** [review]
@ani-sha Good job for geting it running. I just think the new property is not necessary, it is just JDBC driver passthrough property so it does not need to be explicitly added.

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java:614]: Thinking about that as it uses `database.` prefix which is not supported for Debezium 2.x is the config option even needed for 8.2 at all?

Comment (ani-sha) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java]: Yes, I have removed the field. The tests are building now. The test failure seems unrelated.

**Comment by jpechane** [comment]
@ani-sha Applied, thanks

**Pull request merged**